File: ShipmentControlbackend/src/main/java/com/isdmoldova/shipmentcontrolbackend/dto/CargoDTO.java
package com.isdmoldova.shipmentcontrolbackend.dto;
import com.isdmoldova.shipmentcontrolbackend.entity.enums.CargoStatus;

import lombok.Data;
import java.time.LocalDate;
import java.util.List;

@Data
public class CargoDTO {

    private Long id;
    private String trackingNumber;
    private Double totalVolume;
    private Double totalWeight;
    private String origin;
    private String destination;
    private Long currentLegId;
    private CargoStatus cargoStatus;
    private ItineraryDTO itineraryDTO;
    private List<CargoTypeDTO> cargoTypes;
    private LocalDate bookingDate;
    private String user;
    private String provider;
    private String userCompanyName;
    private String providerCompanyName;

}

File: ShipmentControlfrontend/src/app/model/cargo.dto.ts
import {ItineraryDto} from "./itinerary.dto";
import {CargoTypeDto} from "./cargoType.dto";

export interface CargoDto{
    id: number;
    trackingNumber: string;
    totalVolume: number;
    totalWeight: number;
    origin: string;
    destination: string;
    cargoStatus: string;
    itineraryDTO: ItineraryDto;
    cargoTypes: CargoTypeDto[];
    bookingDate: string;
    userCompanyName:string;
    providerCompanyName:string;

}
